providers/cookbook_ci.rb

```diff
@@ -0,0 +1,62 @@
+use_inline_resources if defined?(use_inline_resources)
+
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+#
+# create action
+#
+action :create do
+  job_name = "cookbook-#{new_resource.name}"
+
+  job_config = ::File.join(Chef::Config[:file_cache_path], "#{job_name}-config.xml")
+
+  jenkins_job job_name do
+    action :nothing
+    config job_config
+  end
+
+  commands = ["bundle install --deployment"]
+  commands << "bundle exec rake test:foodcritic" if new_resource.foodcritic
+  commands << "bundle exec rake test:chefspec" if new_resource.chefspec
+  commands << "bundle exec berks" if new_resource.serverspec
```

Comment:
Could the `bundle exec berks` be `bundle exec berks update`? Or perhaps `bundle exec berks verify`? Depending on whether or not we think it is a good idea to change the pins or just to verify they are valid or.. ?